tests/providers/amazon/aws/sensors/test_step_function_execution.py

```diff
@@ -0,0 +1,142 @@
+        hook_response = {
+            'status': 'FAILED'
+        }
+
+        hook_instance = mock_hook.return_value
+        hook_instance.describe_execution.return_value = hook_response
+
+        sensor = StepFunctionExecutionSensor(
+            task_id=TASK_ID,
+            execution_arn=EXECUTION_ARN,
+            aws_conn_id=AWS_CONN_ID,
+            region_name=REGION_NAME
+        )
+
+        with self.assertRaises(AirflowException):
+            sensor.poke(self.mock_context)
+
+    @mock.patch('airflow.providers.amazon.aws.sensors.step_function_execution.StepFunctionHook')
+    def test_timed_out(self, mock_hook):
+        hook_response = {
+            'status': 'TIMED_OUT'
+        }
+
+        hook_instance = mock_hook.return_value
+        hook_instance.describe_execution.return_value = hook_response
+
+        sensor = StepFunctionExecutionSensor(
+            task_id=TASK_ID,
+            execution_arn=EXECUTION_ARN,
+            aws_conn_id=AWS_CONN_ID,
+            region_name=REGION_NAME
+        )
+
+        with self.assertRaises(AirflowException):
+            sensor.poke(self.mock_context)
+
+    @mock.patch('airflow.providers.amazon.aws.sensors.step_function_execution.StepFunctionHook')
+    def test_aborted(self, mock_hook):
+        hook_response = {
+            'status': 'ABORTED'
+        }
+
+        hook_instance = mock_hook.return_value
+        hook_instance.describe_execution.return_value = hook_response
+
+        sensor = StepFunctionExecutionSensor(
+            task_id=TASK_ID,
+            execution_arn=EXECUTION_ARN,
+            aws_conn_id=AWS_CONN_ID,
+            region_name=REGION_NAME
+        )
+
+        with self.assertRaises(AirflowException):
+            sensor.poke(self.mock_context)
+
+    @mock.patch('airflow.providers.amazon.aws.sensors.step_function_execution.StepFunctionHook')
+    def test_running(self, mock_hook):
+        hook_response = {
+            'status': 'RUNNING'
+        }
+
+        hook_instance = mock_hook.return_value
+        hook_instance.describe_execution.return_value = hook_response
+
+        sensor = StepFunctionExecutionSensor(
+            task_id=TASK_ID,
+            execution_arn=EXECUTION_ARN,
+            aws_conn_id=AWS_CONN_ID,
+            region_name=REGION_NAME
+        )
+
+        self.assertFalse(sensor.poke(self.mock_context))
+
+    @mock.patch('airflow.providers.amazon.aws.sensors.step_function_execution.StepFunctionHook')
+    def test_succeeded(self, mock_hook):
+        hook_response = {
+            'status': 'SUCCEEDED'
+        }
+
+        hook_instance = mock_hook.return_value
+        hook_instance.describe_execution.return_value = hook_response
+
+        sensor = StepFunctionExecutionSensor(
+            task_id=TASK_ID,
+            execution_arn=EXECUTION_ARN,
+            aws_conn_id=AWS_CONN_ID,
+            region_name=REGION_NAME
+        )
+
+        self.assertTrue(sensor.poke(self.mock_context))
```

Comment:
Same here.

Reply:
used the @parameterized.expand to remove redundant code and validate True and False returns

Reply:
Oh sorry this one was wrong. Because you are now differ between the status. Can you revert this one? What do you think?

In the other case it makes sense to use paramterized because it is completely the same process - raises exception.

Reply:
i thought the same thing and should have pushed back :). but reverting isn't a problem. i've rolled back that change.